code.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// 全局变量命名不规范
+var global_var int
+var GlobalVar2 string
+
+// 常量命名错误
+const max_SIZE = 100
+const minsize = 10
+
+// 1. 内存泄漏 - goroutine泄漏
+func memoryLeak() {
+	ch := make(chan int)
+	go func() {
+		for {
+			select {
+			case <-ch:
+				return
+			default:
+				// 这个goroutine永远不会退出
+				time.Sleep(time.Second)
+			}
+		}
+	}()
+	// 忘记关闭channel
```

Comment:
Add proper channel cleanup to prevent goroutine leak. The goroutine will exit when the channel is closed.

```suggestion
	// Close channel to properly clean up goroutine
	close(ch)
```